### DISCUSSION TITLE
Navigation beep instead of speech option

### DISCUSSION THREAD

**Discussion by WolfgangZ42** [devemux86/cruiser#570] [Q&A] [answered]
I use the voice announcements both in the Car (via smartphone speaker) and when riding my Motorcycle (via Bluetooth to the helmet communication system) - in both cases just small distance and last announcement, for me perfect .
But: when cycling, I have to use the smartphone speaker as well and that is a bit distracting when we are chatting. Could you imagine just to send a "beep" instead? Would fit perfectly with the screensaver, which I only switch on in the Bike profile. 
Okay - another setup in the voice guidance setting ... 🤔

**Comment by devemux86** [comment]
@WolfgangZ42 thanks for the suggestion.

@0709wiwiwi has also mentioned this feature in the past.

**Reply by 0709wiwiwi** [reply]
Last announcement by tts OR tone. (morse beeps).

_Choose this only as the last announcement alert immediately before the turn. 
If several turns follow shortly after each other, this tone alert restriction is absolutely necessary. 
An excess of multiple tone alerts mixes triggered by different turns is useless and especially confusing. 
Completely unique is to set up a combination of normal TTS with a **SINGLE** tone alert just before the turn._ 

![Panel](https://github.com/user-attachments/assets/a799444e-c39a-49ed-b546-cdaae03c5aa9)

**Left: _  Right: ..  Straight: ...  U-turn: .._ Arrive: ._  and Low beep:  _ for Via and others.**

[ALRSUV.zip](https://github.com/user-attachments/files/18194606/ALRSUV.zip)

**Comment by WolfgangZ42** [comment]
no no no - in my military career I failed the Morse code exam and that was more than 50 years ago.

I just would like the same "beep" on every turn - then I'll look at the screen and see what to do ...

**Reply by devemux86** [reply]
@WolfgangZ42 you mean like a beep warning that a turn is coming in the next X meters.

This is more simple. Like the existing off route beep warning.

**Reply by 0709wiwiwi** [reply]
@WolfgangZ42

>  I failed the Morse code exam and that was more than 50 years ago.

Then you may have practiced too little or too irregularly or had other and more "enchanting" free time interests back then ;-)

**Reply by WolfgangZ42** [reply]
> had other and more "enchanting" free time interests back then ;-) 

You can bet on it! 😏 Btw: my military carrer was quite short and really not impressiv.

**Comment by WolfgangZ42** [comment]
Yesterday I led a 56 km bike tour with Cruiser as head of the gang. Perfect, even on hidden single trails. 
But again I noticed that the announcements for turns were annoying. May I kindly remind you of my suggestion for a “beep” instead? 
To save an additional parameter, could Settings/Voice guidance/Voice volume be used? Like “System” is inserted before 0%, perhaps offer “Beep” after 100%?  

<img width="200" src="https://github.com/user-attachments/assets/3e28a3ef-760f-4fd7-ae4a-9eb63cd80ed6"/>
<img width="200" src="https://github.com/user-attachments/assets/46f10b9a-b19d-48b2-9603-93aec697db2c"/>

Thanks for considering ...

**Reply by devemux86** [reply]
@WolfgangZ42 

Thanks for the reminder!
It is already in my list for implementation. :slightly_smiling_face:

**Comment by MotoUKRider** [comment]
I hadn't looked at this thread, so missed my chance to also suggest that Morse Code would be fine by me :zap: 

Also, consider that with stereo in-helmet speakers there exists the possibility to produce the beep in either the left or right ear, with a sweeping "pan" effect to aid the direction you need to travel in.

*(checks to see if it's actually April 1st)*

**Comment by ToToGSARider** [comment]
But you can then decide whether you want to drive with “BEEP” or with “voice announcement”? I always have the map view running in 3D. Does this issue affect me at all? I always have the screensaver function off, as I haven't yet understood the meaning behind it.

Aber man kann dann schon entscheiden ob man mit "BEEP" fahren möchte oder mit "Sprachansage"? ich habe ja immer die Kartenansicht in 3D laufen. Betrifft mich dann dieses Thema überhaupt? Ich habe die Funktion Bildschirmschoner immer aus, da ich noch nicht den Sinn dahinter verstanden habe.

**Reply by devemux86** [reply]
@ToToGSARider 

> Ich habe die Funktion Bildschirmschoner immer aus, da ich noch nicht den Sinn dahinter verstanden habe.

The screen saver switches the screen off and only opens it when you approach the next turn instruction.

Der Bildschirmschoner schaltet den Bildschirm aus und öffnet ihn erst, wenn Sie sich der nächsten Abbiegeanweisung nähern.

**Comment by devemux86** [comment]
In [Morse code](https://en.wikipedia.org/wiki/Morse_code) there is no distinction between upper and lower case letters.

@0709wiwiwi
In your [document](https://github.com/devemux86/cruiser/discussions/570#discussioncomment-9782611), the only instruction with lower case letter is roundabout? It should also have its own letter, e.g. `R`.
Why complicate our ears with frequencies and same code with Left when there are many free letters?

Also the beeline instruction has no letter?

**Reply by 0709wiwiwi** [reply]
Roundabout 7 variants + via point + more see doc list
The lower tone beep suffices perfectly as an indication.
Beeline is not provided in the Locus sound set.

**Reply by devemux86** [reply]
@0709wiwiwi 

> Roundabout 7 variants + via point + more see doc list

It seems to have errors: same letter for roundabout and via point.

> The lower tone beep suffices perfectly as an indication.

A different code is more clear than same code with different tone.

>  Beeline is not provided in the Locus sound set.

Then Cruiser better use an improved set with different letters for all instructions.

> @WolfgangZ42 even prefers just a simple beep!

I have already implemented a simple beep.
I am checking if Morse code has real value.

**Reply by WolfgangZ42** [reply]
> I have already implemented a simple beep.

Thanks a lot.! Is it already available? 

> I am checking if Morse code has real value.

Personally, I would never use it.

**Reply by devemux86** [reply]
@WolfgangZ42 

> Is it already available?

Beep will be available in Cruiser **5.2**.

**Comment by devemux86** [comment]
Let's go with the simple beep instead of speech option.

1 beep for left, 2 beeps for right, 3 beeps for other.

It will work on the navigation instructions enabled in the voice guidance settings.
So you can customize it and let it only beep at the selected distances, like speech.

<img src="https://github.com/user-attachments/assets/e2a7c5bb-e77b-4941-b904-86ef087cc3f9" width="50%"/>

**Reply by WolfgangZ42** [reply]
I wouldn't want to limit it to the last announcement. My intention is to replace **all** announcements with beep when hiking and cycling and then prepare or initiate the necessary action with a glance at the smartphone.
If I only want to get the beep at the last announcement, I can switch off all other announcements.

**Reply by devemux86** [reply]
@WolfgangZ42 

> I wouldn't want to limit it to the last announcement. My intention is to replace all announcements with beep when hiking and cycling and then prepare or initiate the necessary action with a glance at the smartphone.
If I only want to get the beep at the last announcement, I can switch off all other announcements.

This is the idea, the beep switch is global and replaces the speech.
If an alert (speech or beep) is on / off it can be selected with the same settings as before.

Do we also want the beep to work with the the nearby POI / favorites alerts?

**Reply by WolfgangZ42** [reply]
> Do we also want the beep to work with the the nearby POI / favorites alerts?

Not in my opinion. POI/favorites are special announcements and really don't disturb.

**Reply by devemux86** [reply]
We can use multiple beeps to differentiate left / right instructions, without complicated codes.

**Comment by devemux86** [comment]
Available for testing in:
- https://github.com/devemux86/cruiser/discussions/690

1 beep for left, 2 beeps for right, 3 beeps for other
(Settings | Voice guidance | Advanced)

**Reply by devemux86** [reply]
You can also test the beeps in the simulation of navigation.

Please let me know how you see it or if anything else is needed.

**Reply by WolfgangZ42** [reply]
Thank you for implementing the Beep(s)!
I only want to use it when walking and cycling, but I tested it today when driving. Works like a charm!
Even the pitches are different for left and right, in addition to the 1beep and 2beeps. Perfect!

**Reply by devemux86** [reply]
@WolfgangZ42 Thanks for the feedback! :+1:

**Reply by WolfgangZ42** [reply]
Today I went on my first guided hike with the "beep(s)" - exactly how I had imagined it! Nobody is disturbed by announcements anymore and I still know where I'm going. Thank you once again for the implementation!

**Reply by devemux86** [reply]
@WolfgangZ42 Thanks for the testing and feedback! :+1:

**Comment by devemux86** [comment, marked as answer]
- Navigation beep instead of speech option is available in https://github.com/devemux86/cruiser/discussions/701
